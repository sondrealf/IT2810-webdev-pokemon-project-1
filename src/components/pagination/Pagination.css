.pagination {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
}

.pagination__button {
  height: 70px;
  width: 67px;
  background-color: #fff;
  border: none;
  padding: 0;

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    cursor: default;
    background-color: #e0e0e0;
  }
}

.pagination__button:hover {
  background-color: hsl(5, 70%, 90%);
}

.pagination__left-button {
  border-radius: 10px 0 0 10px;
}

.pagination__right-button {
  border-radius: 0 10px 10px 0;
}

.pagination__index {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
  color: #000;
  background-color: #ffffff;
  width: 70px;
  height: 70px;
  margin: 0 3px;
}
