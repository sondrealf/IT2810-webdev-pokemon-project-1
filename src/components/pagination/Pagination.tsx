import "./Pagination.css";
import ButtonLeft from "../../assets/button-left.svg";
import ButtonRight from "../../assets/button-right.svg";

/**
 * Props for the Pagination component.
 * @interface
 * @property {number} count - The total number of pages.
 * @property {number} currentIndex - The current active page index.
 * @property {(index: number) => void} onChange - A function that is called when the active page index is changed.
 */
interface PaginationProps {
  count: number;
  currentIndex: number;
  onChange: (index: number) => void;
}

/**
 * Renders a pagination component with left and right buttons to navigate between pages.
 * @param {Object} props - The component props.
 * @param {number} props.count - The total number of pages.
 * @param {number} props.currentIndex - The current active page index.
 * @param {Function} props.onChange - The function to be called when the active page index changes.
 * @returns {JSX.Element} - The Pagination component.
 */
const Pagination = ({ count, currentIndex, onChange }: PaginationProps) => {
  return (
    <div className="pagination">
      <button
        className="pagination__button pagination__left-button"
        onClick={() => onChange(currentIndex - 1)}
        disabled={currentIndex === 0}
      >
        <img src={ButtonLeft} alt="Previous" />
      </button>
      <span className="pagination__index">{currentIndex + 1}</span>
      <button
        className="pagination__button pagination__right-button"
        onClick={() => onChange(currentIndex + 1)}
        disabled={currentIndex === count - 1}
      >
        <img src={ButtonRight} alt="Next" />
      </button>
    </div>
  );
};

export default Pagination;
