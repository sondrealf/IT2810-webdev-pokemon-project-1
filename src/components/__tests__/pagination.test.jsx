import { describe, test, expect } from "vitest";
import Pagination from "../pagination/Pagination";
import { render, screen } from "@testing-library/react";

const mockOnChange = vi.fn();

describe("Pagination", () => {
  test("should render", () => {
    render(
      <Pagination
        count={10}
        currentIndex={0}
        onChange={(index) => {
          mockOnChange(index + 1);
        }}
      />
    );
    expect(screen.getByAltText("Previous")).toBeInTheDocument();
    expect(screen.getByAltText("Next")).toBeInTheDocument();
    expect(screen.getByText("1")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <Pagination
        count={10}
        currentIndex={0}
        onChange={(index) => {
          mockOnChange(index + 1);
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("should navigate to next page", () => {
    render(
      <Pagination
        count={10}
        currentIndex={0}
        onChange={(index) => mockOnChange(index + 1)}
      />
    );

    const [previousButton, nextButton] = screen.getAllByRole("button");

    expect(previousButton).toBeDisabled();

    previousButton.click();
    expect(mockOnChange).toHaveBeenCalledTimes(0);

    nextButton.click();
    expect(mockOnChange).toHaveBeenCalledOnce();
    expect(mockOnChange).toHaveBeenCalledWith(2);
  });

  test("should navigate to previous page", () => {
    render(
      <Pagination
        count={10}
        currentIndex={1}
        onChange={(index) => mockOnChange(index + 1)}
      />
    );

    mockOnChange.mockClear();

    const [previousButton, nextButton] = screen.getAllByRole("button");

    previousButton.click();
    expect(mockOnChange).toHaveBeenCalledOnce();
    expect(mockOnChange).toHaveBeenCalledWith(1);
  });
});
