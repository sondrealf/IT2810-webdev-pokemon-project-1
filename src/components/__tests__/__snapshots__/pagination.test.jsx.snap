// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > matches snapshot 1`] = `
<div>
  <div
    class="pagination"
  >
    <button
      class="pagination__button pagination__left-button"
      disabled=""
    >
      <img
        alt="Previous"
        src="/project1/src/assets/button-left.svg"
      />
    </button>
    <span
      class="pagination__index"
    >
      1
    </span>
    <button
      class="pagination__button pagination__right-button"
    >
      <img
        alt="Next"
        src="/project1/src/assets/button-right.svg"
      />
    </button>
  </div>
</div>
`;
